pkg/operator/controller/controller.go

```diff
@@ -415,8 +413,13 @@ func (r *reconciler) ensureIngressController(ci *operatorv1.IngressController, d
 			errs = append(errs, fmt.Errorf("failed to integrate metrics with openshift-monitoring for ingresscontroller %s: %v", ci.Name, err))
 		}
 
-		if err := r.syncIngressControllerStatus(deployment, ci); err != nil {
-			errs = append(errs, fmt.Errorf("failed to sync ingresscontroller status: %v", err))
+		operandEvents := &corev1.EventList{}
+		if err := r.cache.List(context.TODO(), operandEvents, client.InNamespace("openshift-ingress")); err != nil {
+			errs = append(errs, fmt.Errorf("failed to list events in namespace %q: %v", "openshift-ingress", err))
+		} else {
+			if err := r.syncIngressControllerStatus(ci, deployment, lbService, operandEvents.Items); err != nil {
```

Comment:
Seems related to the stuff we've talked about in the area of https://github.com/openshift/cluster-ingress-operator/pull/245#discussion_r291730529 — I do agree with you. The status function should be able to function without the event set in this case.

However, there is a subtle difference between an empty event list from a successful API call and one that's empty because the API call failed. In the latter case, the downstream consumer (status function) has a reduced trust level in the input. Like, if the status function relies on the absence of a value to decide there's a serious problem, should we choose to call the status function with an "undefined" input?

In this case the decision won't affect something serious like availability...

Reply:
I fixed what I think is the core complaint (event lookup failure no longer prevents status sync).